src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -647,6 +647,10 @@ protected virtual bool TryLoadFromCache(JsonOperationContext context)
             SessionInfo sessionInfo = null,
             CancellationToken token = default)
         {
+            if (command.FailoverTopologyEtag == -2)
+            {                
+                command.FailoverTopologyEtag = _nodeSelector?.GetPreferredNodeWithTopology().TopologyEtag ?? -2;
```

Comment:
isn't this the same as `RequestExecutor.TopologyEtag` property?

Reply:
That is not the same, no. Because the topology may change *while* the command is running

Reply:
The source of it is the same. `NodeSelector.Topology` which is taken from `State` inside the NodeSelector.